src/pages/VenueEntrancePage/VenueEntrancePage.tsx

```diff
@@ -20,6 +21,12 @@ export const VenueEntrancePage: React.FunctionComponent<{}> = () => {
   useConnectCurrentVenue();
   const venue = useSelector(currentVenueSelectorData);
 
+  useEffect(() => {
+    if (venue?.showZendesk) {
+      showZendeskWidget();
+    }
+  }, [venue]);
```

Comment:
Should this have an `else` clause where it calls `hideZendeskWidget`? As is, if we show the widget, then the `venue` changes to one where Zendesk should be hidden, I believe it will remain visible.

Also, since this code snippet is repeated in a number of files, there is probably a more DRY way we can implement the pattern. Either by making the check in a 'higher' component in the tree if possible, or at the very least by wrapping up this logic as a custom hook.

Reply:
Backlogged in https://github.com/sparkletown/internal-sparkle-issues/issues/103